### PR TITLE
Expanding Computer Vision for image classification

### PR DESCRIPTION
**Pull request by ryanwelcher**
### Description of the Change
This PR is intended to address the items requested in #75. 

**Items to be addressed:**
- [x] hook to catch the alt text being sent back from the Azure API (in case these need to be presented/filtered by specific consumer integrations)
- [x] Automated tagging
- [x] All automated tasks available to turn on/off via Settings
- [ ] UI for filtering in the Media Library (currently in backbone.js) and then batch sending images for alt text classifying
- [x] UI to add tags alongside automated ones
- [x] ability to remove automated tags
- [x] error messaging or hook available to catch when ClassifAI doesn't match something for an image's alt text

** New items Aug/20/2019**
- [x] Refactor to a single endpoint
- [x] Add taxonomy selector for Image Tagging
- [x] Remove custom taxonomies from Media Library modal UI
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/10up/classifai/blob/develop/CONTRIBUTING.md) document.
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: includes/Classifai/Providers/Azure/ComputerVision.php [hunk 1]
```diff
@@ -160,6 +253,32 @@ public function setup_fields_sections() {
...
+		add_settings_field(
+			'image-tag-threshold',
+			esc_html__( 'Tag Confidence Threshold', 'classifai' ),
+			[ $this, 'render_input' ],
+			$this->get_option_name(),
+			$this->get_option_name(),
+			[
+				'label_for'     => 'tag_threshold',
+				'input_type'    => 'number',
+				'default_value' => 75,
```
Comment (helen) [inline includes/Classifai/Providers/Azure/ComputerVision.php]: We'll want to test to see if this threshold makes sense in some different types of images

Comment (ryanwelcher) [reply includes/Classifai/Providers/Azure/ComputerVision.php]: @helen updated to 70 as the default

File: includes/Classifai/Providers/Azure/ComputerVision.php [hunk 2]
```diff
@@ -8,10 +8,11 @@
 use Classifai\Providers\Provider;
 
 class ComputerVision extends Provider {
+
 	/**
-	 * @var string URL fragment to the describe (caption) API endpoint.
+	 * @var string URL fragment to the analye API endpoint
```
Comment (helen) [inline includes/Classifai/Providers/Azure/ComputerVision.php]:
```suggestion
	 * @var string URL fragment to the analyze API endpoint
```

File: includes/Classifai/Providers/Provider.php [hunk 3]
```diff
@@ -155,6 +155,38 @@ class="<?php echo esc_attr( $class ); ?>"
...
+		$setting_index = $this->get_settings();
+		$saved         = ( isset( $setting_index[ $args['label_for'] ] ) ) ? $setting_index[ $args['label_for'] ] : '';
+		// Check for a default value
+		$saved   = ( empty( $saved ) && isset( $args['default_value'] ) ) ? $args['default_value'] : $saved;
+		$options = isset( $args['options'] ) ? $args['options'] : [];
+		?>
+		<select
+			id="classifai-settings-<?php echo esc_attr( $args['label_for'] ); ?>"
+			name="classifai_<?php echo esc_attr( $this->option_name ); ?>[<?php echo esc_attr( $args['label_for'] ); ?>]"
+			class="<?php echo esc_attr( $class ); ?>"
```
Comment (helen) [inline includes/Classifai/Providers/Provider.php]: I am getting an undefined notice here

**Pull request merged**